picker.go

```diff
@@ -0,0 +1,45 @@
+package grpc
+
+// Picker picks a Conn for RPC requests.
+// This is EXPERIMENTAL and Please do not implement your own Picker for now.
+type Picker interface {
+	// Pick returns the Conn to use for the upcoming RPC. It may return different
+	// Conn's up to the implementation.
+	Pick() (*Conn, error)
+	// Peek returns the Conn use use for the next upcoming RPC. It returns the same
+	// Conn until next time Pick gets invoked.
+	Peek() *Conn
```

Comment:
rewrite the API according to some offline discussion in the team. This API may not be sufficient to support all connection state requirements (e.g., one with many sub-connections).